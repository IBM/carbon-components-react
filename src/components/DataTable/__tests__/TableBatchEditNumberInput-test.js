import React from 'react';
import { shallow } from 'enzyme';
import { TableBatchEditNumberInput } from '../';

describe('DataTable.TableBatchEditNumberInput', () => {
  it('should render', () => {
    const wrapper = shallow(<TableBatchEditNumberInput id="id-foo" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom attributes on the top-level node', () => {
    const wrapper = shallow(
      <TableBatchEditNumberInput id="id-foo" data-foo="Foo" />
    );
    expect(wrapper.prop('data-foo')).toBe('Foo');
  });

  it('should wrap form validation message in a tooltip', () => {
    const wrapper = shallow(
      <TableBatchEditNumberInput
        id="id-foo"
        invalid
        invalidText="invalid-text-foo"
      />
    );
    expect(wrapper.prop('invalidText').props.children).toBe('invalid-text-foo');
  });
});
