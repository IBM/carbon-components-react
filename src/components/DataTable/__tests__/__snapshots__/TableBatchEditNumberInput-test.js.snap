// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.BatchEditableTableNumberInput should render 1`] = `
<NumberInput
  className="bx--number--batch-edit"
  disabled={false}
  helperText=""
  iconDescription="choose a number"
  id="id-foo"
  invalid={false}
  invalidText={<InvalidIcon />}
  label=""
  light={false}
  onChange={[Function]}
  onClick={[Function]}
  step={1}
  value={0}
/>
`;
